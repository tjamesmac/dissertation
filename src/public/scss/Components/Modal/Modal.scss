/**
* Variables
*/
$padding: 15px;
$modalSizing: 150px;

/**
* Mixins
*/
@mixin border {
  border: 1px solid rgba(0, 0, 0, 1);
}
@mixin boxShadow {
  box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.5);
}

/**
* Style
*/

.synonyms {
  position: absolute;
  height: $modalSizing;
  width: $modalSizing;
  background-color: #124673;
  @include border;
  @include boxShadow;

  display: flex;
  justify-content: center;
  flex-direction: column;
  padding: $padding;
  

  .synonyms-item {

    padding: $padding - 10;
    flex: 1 1 auto;
    background-color: #F2A444;

    &:hover {
      background-color: #F2DFCE;
    }
  }

  &:hover {
    // background-color: #F24C27;
  }
}