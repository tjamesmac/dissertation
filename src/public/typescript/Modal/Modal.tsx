import * as React from 'react';
import { IWordAndSynonym } from '../Main/main.interface';

export interface IModalPosition {
  top: number;
  left: number;
}
interface IModalProps {
  words: IWordAndSynonym;
  position: IModalPosition;
  onWordClick: ( event: any ) => void;
}

const Modal = ( prop: IModalProps ): JSX.Element => {

  const words = prop.words;

  const synonyms = words.synonyms;
  const position = prop.position;
  const positionCSS = {
    top: position.top,
    left: position.left,
  };
  const wordMap = synonyms.map(( word: string, index: number ) => {
    return <li
              className='synonyms-item'
              onClick={(e) => prop.onWordClick(e)}
              key={index}>{word}
            </li>;
  });
  return (
    <ul className='synonyms' style={positionCSS}>
      {wordMap}
    </ul>
  );

};

export default Modal;
