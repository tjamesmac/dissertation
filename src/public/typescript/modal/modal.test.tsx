import * as React from 'react';
import renderer from 'react-test-renderer';
import Modal from './modal';

describe('Main page component', () => {

  it('it renders correctly', () => {
    const modalPosition = {top: 1, left: 2};
    const modalWords = { word: 'hello', synonyms: ['hi', 'hiya']};
    const tree = renderer
      .create(<Modal
                hover={() => console.log('test')}
                removeHover={() => console.log('test')}
                words={modalWords}
                position={modalPosition}
                onWordClick={() => console.log('test')}
              />,
              )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});
