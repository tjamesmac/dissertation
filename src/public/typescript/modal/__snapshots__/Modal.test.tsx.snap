// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page component it renders correctly 1`] = `
<ul
  className="synonyms"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "left": 2,
      "top": 1,
    }
  }
/>
`;
